src/docker-extract.c

```diff
@@ -0,0 +1,320 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/file.h>
+#include <sys/stat.h>
+#include <archive.h>
+#include <archive_entry.h>
+
+#include "config.h"
+#include "util/file.h"
+#include "util/message.h"
+#include "util/util.h"
+
+/* apply_opaque
+ *  Given opq_marker as a path to a whiteout opaque marker
+ *    e.g. usr/share/doc/test/.wh..wh..opq
+ *  Make the directory containing the make opaque for this layer by removing it
+ *  if it exists under rootfs_dir
+ */
```

Comment:
Based on the logic here I was expecting directories marked with an opq file to be deleted.  But I see that those directories still exist in a DIGITS image created using this branch.  Moreover in most cases (but not all) the contents of directories marked with a opq file is identical to the contents of the directory if the image is built with the current dist version instead.  What am I missing here?  Is the directory recreated when the current layer is applied?  Is that he expected/desired behavior?

Reply:
An opaque directory is one that was removed, but then recreated later. It`s opaque in that the files from earlier layers don’t show through. Believe I’m implementing this correctly. Directory is remove when processing the white outs then will reappear when layer tar is extracted.

Reply:
OK.  As long as it is working as expected. Thanks for clarification. 😸